src/main/webapp/app/entities/result/result.component.ts

```diff
@@ -121,6 +195,27 @@ export class ResultComponent implements OnInit, OnChanges {
         return this.result!.hasFeedback;
     }
 
+    hasParticipationResults(): boolean {
+        if (!this.participation) {
+            return false;
+        }
+        return this.participation.results && this.participation.results.length > 0;
+    }
+
+    isSubmissionInDueTime(): boolean {
```

Comment:
- Undocumented, complicated function.
- Should be a util
- No inline doc for the conditional expressions
- Before doing anything at all, the first evaluation should be:

```
// If the exercise has no dueDate set, every submission is in time.
if(!this.participation.exercise.dueDate){
  return true;
} 
```

Reply:
- created a util and moved it there
- documented the function
- adjusted the code as you suggested